.github/workflows/gotidy.yml

```diff
@@ -0,0 +1,29 @@
+# This action requires that PRs run `make gotidy` to ensure
+# dependencies are up to date in reaction to any changes
+
+name: gotidy
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches:
+      - main
+
+jobs:
+  gotidy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Check for go mod dependency changes
+        run: |
+          make gotidy
+          if [[ $(git diff --name-only | grep go\.mod) || $(git diff --name-only | grep go\.sum) ]]
```

Comment:
Does this workflow fail any time go.mod or go.sum are modified for any reason?

Reply:
It shouldn't, the intent is to run gotidy against the head of the PR branch and detect if any new changes in go.mod/sum appear. Not trying to `git diff` against upstream/main

Reply:
I understand now.  Assuming `make gotidy` was already run and committed, the run as part of this workflow should not change anything.